mmhuman3d/core/visualization/renderer/torch3d_renderer/utils.py

```diff
@@ -0,0 +1,114 @@
+from typing import List, Union
+
+import numpy as np
+import torch
+from pytorch3d.structures import list_to_padded
+
+try:
+    from typing import Literal
+except ImportError:
+    from typing_extensions import Literal
+
+
+def normalize(value,
+              origin_value_range=None,
+              out_value_range=(0, 1),
+              dtype=None,
+              clip=False) -> Union[torch.Tensor, np.ndarray]:
+    """Normalize the tensor or array and convert dtype."""
+    if origin_value_range is not None:
+        value = (value - origin_value_range[0]) / (
+            origin_value_range[1] - origin_value_range[0] + 1e-9)
+
+    else:
+        value = (value - value.min()) / (value.max() - value.min())
+    value = value * (out_value_range[1] -
+                     out_value_range[0]) + out_value_range[0]
+    if clip:
+        value = torch.clip(
+            value, min=out_value_range[0], max=out_value_range[1])
+    if isinstance(value, torch.Tensor):
+        if dtype is not None:
+            return value.type(dtype)
+        else:
+            return value
+    elif isinstance(value, np.ndarray):
+        if dtype is not None:
+            return value.astype(dtype)
+        else:
+            return value
+
+
+def tensor2array(image: torch.Tensor) -> np.ndarray:
+    """Convert image tensor to array."""
+    image = image.detach().cpu().numpy()
+    image = normalize(
+        image,
+        origin_value_range=(0, 1),
+        out_value_range=(0, 255),
+        dtype=np.uint8)
+    return image
+
+
+def array2tensor(image: np.ndarray) -> torch.Tensor:
+    """Convert image array to tensor."""
+    image = torch.Tensor(image)
+    image = normalize(
+        image,
+        origin_value_range=(0, 255),
+        out_value_range=(0, 1),
+        dtype=torch.float32)
+    return image
+
+
+def rgb2bgr(rgbs) -> Union[torch.Tensor, np.ndarray]:
+    """Convert color channels."""
+    bgrs = [rgbs[..., 0, None], rgbs[..., 1, None], rgbs[..., 2, None]]
```

Comment:
It should be `bgrs = [rgbs[..., 2, None], rgbs[..., 1, None], rgbs[..., 0, None]]`. Otherwise, no channel has been switched.

Reply:
ok